core/planner.c

```diff
@@ -1072,10 +1074,17 @@ bool plan(struct deco_state *ds, struct diveplan *diveplan, struct dive *dive, i
 	if (prefs.surface_segment != 0) {
 		// Switch to an empty air cylinder for breathing air at the surface
 		// If no empty cylinder is found, keep using last deco gas
+		// FIXME: This is incredibly silly and emulates the old code when
+		// we had a fixed cylinder table: It makes sure that there is at least
+		// one cylinder beyond the used cylinders in the cylinder table.
+		// It then lets the cylinder point past the end of the table. Thus,
+		// this fake-cylinder can be used by the planner, but is not visible to
+		// the UI. Fix this as soon as possible!
 		cylinder_t cyl = empty_cylinder;
 		cyl.cylinder_use = NOT_USED;
 		add_to_cylinder_table(&dive->cylinders, dive->cylinders.nr, cyl);
 		current_cylinder = dive->cylinders.nr - 1;
+		dive->cylinders.nr--;
```

Comment:
As I said, I would vote for always having an extra 0 cylinder at the end, so we are not decreasing the number in every run but simply use that last cylinder.